tests/lib/k8s/test_builders.py

```diff
@@ -1300,3 +1311,30 @@ def test_get_sections_for_cronjob_with_namespace(
         set_environment_variables_mock.assert_called_once()
         set_node_selector_mock.assert_called_once()
         set_namespace_mock.assert_called_once()
+
+    def test_build_ingress_yaml(self):
+        # Mocks
+        deployment_config = {
+            "app": "my-app",
+            "domain": "my-app.com",
+            "domain_prefix": "-staging",
+        }
+        templates = {
+            "ingress-app": self._create_template_validator(
```

Comment:
Maybe it would be good to have all expectation use this regarding templates and remove `check_hpa`, `check_namespace`, etc.